test/multiple-user-principal-types.test.js

```diff
@@ -0,0 +1,978 @@
+      });
+    });
+
+    it('logs in a user from user model 2 with realm', function(done) {
+      User2.login(credentialWithRealm, function(err, accessToken) {
+        assertGoodToken(accessToken, user1b);
+
+        done();
+      });
+    });
+
+    it('logs in a user by username with realm', function(done) {
+      User.login(credentialRealmInUsername, function(err, accessToken) {
+        assertGoodToken(accessToken, user1a);
+
+        done();
+      });
+    });
+
+    it('logs in a user by email with realm', function(done) {
+      User.login(credentialRealmInEmail, function(err, accessToken) {
+        assertGoodToken(accessToken, user1a);
+
+        done();
+      });
+    });
+
+    describe('User.login with realmRequired but no realmDelimiter', function() {
+      beforeEach(function() {
+        User.settings.realmDelimiter = undefined;
+      });
+
+      afterEach(function() {
+        User.settings.realmDelimiter = ':';
+      });
+
+      it('logs in a user by with realm', function(done) {
+        User.login(credentialWithRealm, function(err, accessToken) {
+          assertGoodToken(accessToken, user1a);
+
+          done();
+        });
+      });
+
+      it('rejects a user by with realm in email if realmDelimiter is not set',
+        function(done) {
+          User.login(credentialRealmInEmail, function(err, accessToken) {
+            assert(err);
+            assert.equal(err.code, 'REALM_REQUIRED');
+
+            done();
+          });
+        });
+    });
+  });
+
+  describe('User.logout', function() {
+    it('Logout a user by providing the current accessToken id (using node)', function(done) {
+      login(logout);
+
+      function login(fn) {
+        User.login({email: 'foo@bar.com', password: 'bar'}, fn);
+      }
+
+      function logout(err, accessToken) {
+        User.logout(accessToken.id, verify(accessToken.id, done));
+      }
+    });
+
+    it('Logout a user by providing the current accessToken id (using node) - promise variant',
+    function(done) {
+      login(logout);
+
+      function login(fn) {
+        User.login({email: 'foo@bar.com', password: 'bar'}, fn);
+      }
+
+      function logout(err, accessToken) {
+        User.logout(accessToken.id)
+          .then(function() {
+            verify(accessToken.id, done);
+          })
+          .catch(done(err));
+      }
+    });
+
+    function verify(token, done) {
+      assert(token);
+
+      return function(err) {
+        if (err) return done(err);
+
+        AccessToken.findById(token, function(err, accessToken) {
+          assert(!accessToken, 'accessToken should not exist after logging out');
+
+          done(err);
+        });
+      };
+    }
+
+    it('Logout a user from user model 1 without logout user from model 2', function(done) {
+      User.login({email: 'foo@bar.com', password: 'bar'}, function(err, accessToken1) {
+        User.login({email: 'foo@bar.com', password: 'bar'}, function(err, accessToken2) {
```

Comment:
This is using the same `User` model as in the first token, shouldn't it be `User2` instead?